flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer.java

```diff
@@ -516,10 +510,12 @@ public void close() throws Exception {
 
 		if (LOG.isDebugEnabled()) {
 			LOG.debug("Snapshotting state. Offsets: {}, checkpoint id: {}, timestamp: {}",
-					Arrays.toString(lastOffsets), checkpointId, checkpointTimestamp);
+					KafkaTopicPartition.toString(lastOffsets), checkpointId, checkpointTimestamp);
 		}
 
-		long[] currentOffsets = Arrays.copyOf(lastOffsets, lastOffsets.length);
+		// the use of clone() is okay here is okay, we just need a new map, the keys are not changed
+		//noinspection unchecked
+		HashMap<KafkaTopicPartition, Long> currentOffsets = (HashMap<KafkaTopicPartition, Long>) lastOffsets.clone();
```

Comment:
Clone is probably a bit more efficient even, as does not hash but only copy the table.